Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml

```diff
@@ -13,29 +13,131 @@ triggerOperator: gt
+  // https://docs.aws.amazon.com/guardduty/latest/APIReference/API_KubernetesApiCallAction.html
+  | extend RemoteIpAddress = 
+      coalesce(
+        tostring(ServiceAction.remoteIpDetails.ipAddressV4),
+        tostring(parse_json(ServiceAction.sourceIPs)[0])
+      )
+  // The IPv4 local address of the connection
+  // https://docs.aws.amazon.com/guardduty/latest/APIReference/API_LocalIpDetails.html
+  | extend LocalIpAddress = ServiceAction.localIpDetails.ipAddressV4
+  // The AWS account ID of the remote API caller.
+  // https://docs.aws.amazon.com/guardduty/latest/APIReference/API_AwsApiCallAction.html
+  // https://docs.aws.amazon.com/guardduty/latest/APIReference/API_RemoteAccountDetails.html
+  | extend RemoteAWSAccountId = ServiceAction.remoteAccountDetails.accountId
+  // The IAM access key details (user information) of a user that engaged in the activity that prompted GuardDuty to generate a finding
+  // https://docs.aws.amazon.com/guardduty/latest/APIReference/API_AccessKeyDetails.html
+  | extend AccountUpn = 
+      case(
+        AccessKeyDetails.userType == "IAMUser", AccessKeyDetails.userName,
+        AccessKeyDetails.userType == "AssumedRole", split(AccessKeyDetails.principalId, ":", 1)[0],
+        isnotempty(RdsDbUserDetails.user), RdsDbUserDetails.user,
+        isnotempty(KubernetesDetails.kubernetesUserDetails.username), KubernetesDetails.kubernetesUserDetails.username,
+        ""
+      )
+  | extend AccountName = split(AccountUpn, "@", 0)[0]
+  | extend UPNSuffix = split(AccountUpn, "@", 1)[0]
+  // Clean up the output
+  | extend GuardDutyDetails =
+      bag_pack( 
+        "DetectorId", ServiceDetails.detectorId,
+        "Partition", Partition,
+        "Region", Region
+      )
+  | extend FindingLink = 
+      iff(
+        isnotempty(Region) and isnotempty(Id),
+        strcat("https://", Region, ".console.aws.amazon.com/guardduty/home?region=", Region, "#/findings?fId=", Id),
+        ""
+      )
+  | extend FindingLinkDescription = 
+      iff(
+        isnotempty(FindingLink),
+        strcat("Link to GuardDuty finding (AWS): ", FindingLink),
+        ""
+      )
+  | project-rename 
+      FindingArn=Arn,
+      FindingId=Id,
+      AWSAccountId=AccountId
+  | project-away 
+      ActivityType, 
+      findingTokens,
+      Partition,
+      Region, 
+      SchemaVersion,
+      TimeGenerated,
+      Type
+version: 1.0.3
 entityMappings:
   - entityType: Account
     fieldMappings:
       - identifier: Name
-        columnName: Arn
+        columnName: AccountName
+      - identifier: UPNSuffix
+        columnName: UPNSuffix
       - identifier: ObjectGuid
-        columnName: AWSAcoundId
+        columnName: RemoteAWSAccountId
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: RemoteIpAddress
+      - identifier: Address
+        columnName: LocalIpAddress
```

Comment:
Two values are provided for same Identifier. We would have to create a new block for IP entity if we need to map two IPs

Reply:
Fixed